### PR TITLE
use seconds for all duration with same feature flag we already have

### PR DESCRIPTION
**Pull request by zeitlinger**
Depends on https://github.com/open-telemetry/semantic-conventions/pull/176

### CODE REVIEW

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/metrics/DurationHistogram.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.metrics;
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/metrics/DurationHistogram.java:6]: Can you place it in a `...metrics.internal` package? I don't think we want this to become a part of the public  `instrumentation-api` API

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/metrics/DurationHistogram.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.concurrent.TimeUnit;
+import javax.annotation.concurrent.ThreadSafe;
+
+/** A Histogram instrument that records durations */
+@ThreadSafe
+public class DurationHistogram {
+
+  private final DoubleHistogram delegate;
+
+  public DurationHistogram(DoubleHistogram delegate) {
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/metrics/DurationHistogram.java:20]: Could be package private
```suggestion
  DurationHistogram(DoubleHistogram delegate) {
```

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/metrics/DurationHistogram.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.context.Context;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.concurrent.ThreadSafe;
+
+/** A Histogram instrument that records durations */
+@ThreadSafe
+public class DurationHistogram {
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/metrics/DurationHistogram.java:16]:
```suggestion
public final class DurationHistogram {
```

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/metrics/DurationHistogramFactory.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import static java.util.Collections.unmodifiableList;
+
+import io.opentelemetry.api.metrics.DoubleHistogramBuilder;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.extension.incubator.metrics.ExtendedDoubleHistogramBuilder;
+import io.opentelemetry.instrumentation.api.internal.SemconvStability;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class DurationHistogramFactory {
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/metrics/DurationHistogramFactory.java:18]:
```suggestion
public final class DurationHistogramFactory {
```

File: instrumentation/hikaricp-3.0/library/src/main/java/io/opentelemetry/instrumentation/hikaricp/v3_0/OpenTelemetryMetricsTracker.java [hunk 5]
```diff
@@ -44,20 +42,19 @@ final class OpenTelemetryMetricsTracker implements IMetricsTracker {
...
-    double millis = elapsedAcquiredNanos / NANOS_PER_MS;
-    waitTime.record(millis, attributes);
+    waitTime.record(elapsedAcquiredNanos, TimeUnit.NANOSECONDS, attributes);
     userMetricsTracker.recordConnectionAcquiredNanos(elapsedAcquiredNanos);
   }
 
   @Override
   public void recordConnectionUsageMillis(long elapsedBorrowedMillis) {
-    useTime.record((double) elapsedBorrowedMillis, attributes);
+    useTime.record(elapsedBorrowedMillis, TimeUnit.MICROSECONDS, attributes);
```
Comment (mateuszrzeszutek) [inline instrumentation/hikaricp-3.0/library/src/main/java/io/opentelemetry/instrumentation/hikaricp/v3_0/OpenTelemetryMetricsTracker.java:57]:
```suggestion
    useTime.record(elapsedBorrowedMillis, TimeUnit.MILLISECONDS, attributes);
```

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/metrics/internal/DurationHistogramFactory.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import static java.util.Collections.unmodifiableList;
+
+import io.opentelemetry.api.metrics.DoubleHistogramBuilder;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.extension.incubator.metrics.ExtendedDoubleHistogramBuilder;
+import io.opentelemetry.instrumentation.api.internal.SemconvStability;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public final class DurationHistogramFactory {
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/metrics/internal/DurationHistogramFactory.java:18]: You'll need to add the following comment to public internal classes to fix the build:
```suggestion
/**
 * This class is internal and is hence not for public use. Its APIs are unstable and can change at
 * any time.
 */
public final class DurationHistogramFactory {
```

**Comment by trask** [comment]
@zeitlinger is this PR still needed? thx

**Comment by zeitlinger** [comment]
there was a nice refactoring in it, but I didn't separate it out.

**Pull request closed without merging**